src/main/java/net/citizensnpcs/EventListen.java

```diff
@@ -559,4 +570,45 @@ public int hashCode() {
             return prime * (prime * (prime + ((worldName == null) ? 0 : worldName.hashCode())) + x) + z;
         }
     }
+
+    private class SkinUpdateTracker {
+        float initialYaw;
+        final Location location = new Location(null, 0, 0, 0);
+        boolean hasMoved;
+        int rotationCount;
+
+        SkinUpdateTracker(Player player) {
+            reset(player);
+        }
+
+        boolean shouldUpdate(Player player) {
+
+            if (!hasMoved) {
+                hasMoved = true;
+                return true;
+            }
+
+            Location currentLoc = player.getLocation(YAW_LOCATION);
+            float currentYaw = currentLoc.getYaw();
+
+            boolean hasRotated = Math.abs(currentYaw - this.initialYaw) < 90;
```

Comment:
The larger the constant is the more likely it is to update right? That's the way I read it.